stable_plugins/visualization/file_types/qasm_visualization_templates/qasm_visualization.html

```diff
@@ -0,0 +1,102 @@
+{% import 'forms.html' as forms %}
+
+<html>
+
+<head>
+    <link rel="stylesheet" type="text/css" href="{{url_for('static', filename='forms.css')}}" media="screen" />
+    <style>
+        .qhana-data-preview {
+            text-rendering: optimizeSpeed;
+            display: flex;
+            flex-direction: row;
+            flex-wrap: wrap;
+            max-width: 100%;
+            max-height: 750px;
+            overflow: auto;
+            background: transparent;
+        }
+
+        #spinner {
+            display: inline-block;
+            width: 100px;
+            height: 100px;
+            border: 6px solid rgba(0, 0, 0, 0);
+            border-radius: 50%;
+            border-top-color: #673ab7;
+            animation: spin 1s linear infinite;
+            -webkit-animation: spin 1s linear infinite;
+        }
+
+        .container {
+            display: flex;
+            justify-content: center;
+            align-items: center;
+            width: 100%;
+        }
+
+        @keyframes spin {
+            to {
+                -webkit-transform: rotate(360deg);
+            }
+        }
+    </style>
+</head>
+
+<body>
+    <div class="qhana-microfrontend">
+        <details id="visualization-options" open>
+            <summary>Visualization Options</summary>
+            {% call forms.render_form(target="microfrontend") %}
+            {{ forms.render_fields(schema, values=values, errors=errors) }}
+            <div class="qhana-form-buttons">
+                {{ forms.submit("validate", target="microfrontend") }}
+            </div>
+            {% endcall %}
+        </details>
+    </div>
+
+    <img id="circuit-image" class="qhana-data-preview" onload="onCircuitImageLoad()" onerror="onCircuitImageError()">
```

Comment:
Do not set event listeners like this in HTML! This can be a security risk so webpages can be configured to disallow this. Instead, use `<element>.addEventListener('load', onCircuitImageLoad)` from your JavaScript to add the listeners and `document.getElementById` or `document.querySelector` to find your element.